### PR TITLE
DateInput UI

### PR DESCRIPTION
**Pull request by sarajaur**
# Related Issue

A Pull Request should close a **single** issue; multiple issues can be closed when the issues are **small and related**, but that should be an exception not the rule. Please keep Pull Requests small and targeted; large 'code drops' with dozens of files will be closed and asked to split into reviewable pieces. Reviews that need to be large due to dependencies will be
reviewed on a case-by-case basis.

Please use one of the well-known [github fixes keywords](https://help.github.com/en/articles/closing-issues-using-keywords) to reference
the issue fixed with this PR (eg Fixes #<github issue number>). If an issue doesn't yet exist please create one to aid
in issue tracking.

**NOTE**: For multiple issues resolved by this PR use the corresponding keywords **every time** in a comma-delimited list per the reference
page above.

# Description

For all Pull Requests, please describe how the issue was fixed or how the feature was implemented from a summary level. This information will be used to help provide context to the reviewers of the pull request and should be additive to the issue being closed.

- Implemented Date Input UI for the Model Base approach.
- 
image

# Sample Card

If appropriate, please include a link to a card in one of the samples directories that can be used to validate this change. This can be an existing card or a card added with this PR.

# How Verified

How you verified the fix, including one or all of the following:
1. New unit tests that were added if any. If none were added please add a quick line explaining why not.
2. Existing relevant unit/regression tests that you ran
3. Manual scenario verification if any; ***Do include .gif's or screenshots of the testing you performed here if you think that it
will aid in code reviews or corresponding fixes on other platforms for eg.***

### CODE REVIEW

File: source/qml_v2/AdaptiveCardQmlEngine/models/DateInputModel.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#include "DateInputModel.h"
+#include "SharedAdaptiveCard.h"
+#include <QDebug.h>
+#include "Utils.h"
+#include "MarkDownParser.h"
+
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/models/DateInputModel.cpp:6]: Are we using this ?

File: source/qml_v2/AdaptiveCardQmlEngine/models/DateInputModel.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#pragma once
+#include "AdaptiveCardContext.h"
+#include "DateInput.h"
+#include "OpenUrlAction.h"
+#include "ToggleVisibilityAction.h"
+#include "SubmitAction.h"
+
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/models/DateInputModel.h:7]: Are we using these actions  ?

File: source/qml_v2/AdaptiveCardQmlEngine/qml/AdaptiveCard.qml [hunk 3]
```diff
@@ -96,6 +96,7 @@ Rectangle {
     onCardThemeChanged: {
         adaptiveCardController.setCardTheme(cardTheme)
         adaptiveCardController.setCardJSON(cardJSON)
+
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/qml/AdaptiveCard.qml:99]: Why this change ?

File: source/qml_v2/AdaptiveCardQmlEngine/qml/DateInputPopout.qml [hunk 4]
```diff
@@ -0,0 +1,332 @@
+﻿import QtQuick 2.15
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/qml/DateInputPopout.qml:1]: Is this file required fir this PR ?

File: source/qml_v2/AdaptiveCardQmlEngine/qml/InputErrorMessage.qml [hunk 5]
```diff
@@ -1,12 +1,15 @@
 import QtQuick 2.15
 import QtQuick.Layouts 1.3
+import QtGraphicalEffects 1.15
 import QtQuick.Controls 2.15
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/qml/InputErrorMessage.qml:4]: What is this used for ?

File: source/qml_v2/AdaptiveCardQmlEngine/qml/InputFieldClearIcon.qml [hunk 6]
```diff
@@ -1,12 +1,13 @@
 import QtQuick 2.15
 import QtQuick.Layouts 1.3
+import QtGraphicalEffects 1.15
 import QtQuick.Controls 2.15
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/qml/InputFieldClearIcon.qml:4]: What is this used for ?

File: source/qml_v2/AdaptiveCardQmlEngine/qml/TextInputRender.qml [hunk 7]
```diff
@@ -118,7 +118,6 @@ Column {
     
     InputErrorMessage {
         id: inputtextErrorMessage
-        
         visible: showErrorMessage
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/qml/TextInputRender.qml:121]: Why this change ?

File: source/qml_v2/AdaptiveCardQmlEngine/models/DateInputModel.cpp (hunk 1 above)
Comment (sarajaur) [reply source/qml_v2/AdaptiveCardQmlEngine/models/DateInputModel.cpp:6]: yes

File: source/qml_v2/AdaptiveCardQmlEngine/qml/AdaptiveCard.qml (hunk 3 above)
Comment (sarajaur) [reply source/qml_v2/AdaptiveCardQmlEngine/qml/AdaptiveCard.qml:99]: There is no changes only extra space was there. I'll remove it.

File: source/qml_v2/AdaptiveCardQmlEngine/qml/DateInputPopout.qml (hunk 4 above)
Comment (sarajaur) [reply source/qml_v2/AdaptiveCardQmlEngine/qml/DateInputPopout.qml:1]: yes

File: source/qml_v2/AdaptiveCardQmlEngine/qml/InputErrorMessage.qml (hunk 5 above)
Comment (sarajaur) [reply source/qml_v2/AdaptiveCardQmlEngine/qml/InputErrorMessage.qml:4]: This is related to the date image icon.

File: source/qml_v2/AdaptiveCardQmlEngine/qml/InputFieldClearIcon.qml (hunk 6 above)
Comment (sarajaur) [reply source/qml_v2/AdaptiveCardQmlEngine/qml/InputFieldClearIcon.qml:4]: This is related to all the icon images.

File: source/qml_v2/AdaptiveCardQmlEngine/models/DateInputModel.h (hunk 2 above)
Comment (sarajaur) [reply source/qml_v2/AdaptiveCardQmlEngine/models/DateInputModel.h:7]: No

File: source/qml_v2/AdaptiveCardQmlEngine/models/DateInputModel.h [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+#include <QString>
+#include <QColor>
+#include <QFont>
+
+class DateInputModel : public QObject
+{
+    Q_OBJECT
+
+   
+
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/models/DateInputModel.h:18]: Extra newline ?

File: source/qml_v2/AdaptiveCardQmlEngine/qml/TextInputRender.qml (hunk 7 above)
Comment (sarajaur) [reply source/qml_v2/AdaptiveCardQmlEngine/qml/TextInputRender.qml:121]: There was no changes in this file. this change happened by mistake.

**Review by soumykun — APPROVED** [review]

**Review by ramkrishnajana — APPROVED** [review]
